platform/admin/core/admin-core-alertservice-data/src/main/java/org/codice/ddf/admin/core/alert/service/data/internal/Alert.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.core.alert.service.data.internal;
+
+import static org.apache.commons.lang.Validate.notEmpty;
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.util.HashMap;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import net.shibboleth.utilities.java.support.annotation.constraint.NotEmpty;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library.</b>
+ * </p>
+ * <p>
+ * An {@link Alert} holds the content for a system administrator alert.
+ */
+public class Alert extends HashMap<String, Object> {
+
+    public static final String ALERT_EVENT_TOPIC_VALUE = "ddf/alertservice";
+
+    public static final String ALERT_EVENT_ALERT_PROPERTY = "alert";
+
+    private static final String LEVEL = "level";
+
+    private static final String TITLE = "title";
+
+    private static final String DETAILS = "details";
+
+    private static final String KEY = "key";
+
+    /**
+     * Constructs an {@link Alert} with a {@param key}
+     *
+     * @param key     non-null, non-empty {@value #KEY}
```

Comment:
Expounding on @lessarderic's comment above, we always assume `null` *cannot* be returned unless the `@Nullable` annotation is present.